dds/DCPS/transport/framework/TransportDefs.h

```diff
@@ -249,6 +249,26 @@ static const ConnectionInfoFlags CONNINFO_UNICAST = (1 << 0);
 static const ConnectionInfoFlags CONNINFO_MULTICAST = (1 << 1);
 static const ConnectionInfoFlags CONNINFO_ALL = static_cast<ConnectionInfoFlags>(-1);
 
+struct RelayMessageCounts {
```

Comment:
It feels a bit messy to have RelayMessageCounts get passed down into the transport where stun_send and stun_recv are ignored. Maybe RelayMessageCounts (w/ stun_send, stun_recv) inherits from Basic(Rtps)MessageCounts (w/ (rtps_)send & recv)? And then a BasicMessageCount reference could get passed down into the transport with a `get_and_reset_message_counts` method??

Reply:
Ideally, this wouldn't be in the framework at all.
However, the way the dependencies work and the design of the transport framework forced me to put them here.
Removing the stun members to a derived class obviously means that there would be some form of casting in the transport.  I think we wait to see if fixing the framework makes some of these unnecessary and/or another transport to make use of the feature.